### PR TITLE
Sorted out the projectile mess

### LINKED ISSUE #1192: ProjectileEntity extends Projectile extends Entity
This is confusing...

### PR DESCRIPTION
**Pull request by YanisBft**
Projectile inheritance was changed in one of the 1.16 snapshots. This PR sorts out all the mess created by this change.
Instead of long explanations, here's two schemas showing the changes. Note that some entity classes were renamed to match their actual IDs, such as thrown projectiles.
This PR closes #1192 

Before:
before

After:
after

### CODE REVIEW

File: mappings/net/minecraft/entity/EyeOfEnderEntity.mapping [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_1672 net/minecraft/entity/EnderEyeEntity
+CLASS net/minecraft/class_1672 net/minecraft/entity/EyeOfEnderEntity
```
Comment (Yoghurt4C) [inline mappings/net/minecraft/entity/EyeOfEnderEntity.mapping:1]: please no

**Review by Yoghurt4C — COMMENTED** [review]
please no

File: mappings/net/minecraft/entity/EyeOfEnderEntity.mapping (hunk 1 above)
Comment (YanisBft) [reply mappings/net/minecraft/entity/EyeOfEnderEntity.mapping:1]: accept mojang inconsistency yog :irritatered:

**Review by Runemoro — APPROVED** [review]

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/entity/EyeOfEnderEntity.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/entity/EyeOfEnderEntity.mapping:1]: With an anime Minecraft skin, Yoghurt4C's opinion doesn't count

Comment (Yoghurt4C) [reply mappings/net/minecraft/entity/EyeOfEnderEntity.mapping:1]: why do I have to suffer because of 🇫🇷 terrible frenchmen 🇫🇷

**Review by Yoghurt4C — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Pull request merged**